### ISSUE TITLE
Accessibility: Buttons incorrectly given the role of links

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51896]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

Buttons incorrectly have <a> tags on them, though they do not function as links and are not redirecting to another URL. Buttons are elements that trigger an action on a page (opening a lightbox, expanding an accordion, submitting a form).

![Image](https://github.com/user-attachments/assets/60ec4bc7-b277-4795-b802-5157d8c1d8f0)

### Component or Page: 

Shortcode: [woocommerce_cart] & Cart Classic Block

### Name affected element:

Change address button

### Relevant code snippets:

`<a href="#" class="shipping-calculator-button">Change address</a>`

### Recommended fix:

Use the appropriate <button> tag on the elements.
 
If you cannot replace a <a> tag with a button, it is possible to remediate a link to function as a button with ARIA. The link would need to be given a button role and href=""#"" to ensure that it can be focused with the keyboard. You would also need to write JavaScript to ensure that the link will function properly as a button for keyboard users and can be triggered with the space bar in addition to the return/enter key. For more information on turning a link into a button, see: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by pierorocca** [comment]
Hi @vikrampm1 - thanks for the report.

I know this has been open a while, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and marked this ready for the engineering team to pick up.

What I'm seeing: Latest audit evidence: Open PR metadata covers this issue; Cart.

Team there's an unactioned PR from 2024.
